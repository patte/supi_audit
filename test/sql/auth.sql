begin;
    -- disable logging to avoid cluttering the test output
    \set ECHO none
    set client_min_messages = warning;

    -- begin before audit is created
    drop schema if exists audit cascade;

    create schema auth;

    create function auth.uid()
        returns uuid
        language sql
    as $$ select '76f99606-1b3f-41d1-806d-358b34db3b32'::uuid $$;

    create function auth.role()
        returns text
        language sql
    as $$ select 'anon' $$;

    -- load the audit migration
    \i test/fixtures.sql

    -- re-enable logging
    set client_min_messages = notice;
    \set ECHO all

    -- Check that the auth_uid and auth_role columns are present
    select * from audit.record_version;

    create table public.xyz(id int primary key);

    select audit.enable_tracking('public.xyz'::regclass);

    insert into public.xyz(id) values(1);

    -- Check that defaults populate
    select
        op,
        auth_uid,
        auth_role
    from
        audit.record_version;

rollback;
