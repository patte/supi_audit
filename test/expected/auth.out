begin;
    -- disable logging to avoid cluttering the test output
    \set ECHO none
    -- Check that the auth_uid and auth_role columns are present
    select id, auth_uid, auth_role from audit.record_version;
 id | auth_uid | auth_role 
----+----------+-----------
(0 rows)

    create table public.xyz(id int primary key);
    select audit.enable_tracking('public.xyz'::regclass);
 enable_tracking 
-----------------
 
(1 row)

    insert into public.xyz(id) values(1);
    -- Check that defaults populate
    select
        op,
        auth_uid,
        auth_role
    from
        audit.record_version;
   op   |               auth_uid               | auth_role 
--------+--------------------------------------+-----------
 INSERT | 76f99606-1b3f-41d1-806d-358b34db3b32 | anon
(1 row)

rollback;
